### PR TITLE
fix(svg): fix chart cannot be exported with SVG renderer in IE.

### PR DESCRIPTION
**Pull request by plainheart**
## Brief Information

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others

### What does this PR do?

**1) fix charts cannot be exported when using SVG renderer in IE.**

**2) fix exported SVG file cannot be opened when using external images.**
See #13655

After this fix, exporting SVG file can even be supported in IE 9.

### Fixed issues

- #13655

### Are there any API changes?

- [ ] The API has been changed.

### Related test cases or examples to use the new APIs

Please refer to `test/ie-svg-export.html`.

### Merging options

- [ ] Please squash the commits into a single one when merge.

### Other information

Related PR in zrender ecomfe/zrender#673

### CODE REVIEW

File: src/component/toolbox/feature/SaveAsImage.ts [hunk 1]
```diff
@@ -73,15 +73,42 @@ class SaveAsImage extends ToolboxFeature<ToolboxSaveAsImageFeatureOption> {
...
-            if (window.navigator.msSaveOrOpenBlob) {
-                const bstr = atob(url.split(',')[1]);
-                let n = bstr.length;
-                const u8arr = new Uint8Array(n);
-                while (n--) {
-                    u8arr[n] = bstr.charCodeAt(n);
+            if (window.navigator.msSaveOrOpenBlob || isSvg) {
+                const parts = url.split(',');
+                // data:[<mime type>][;charset=<charset>][;base64],<encoded data>
+                // see https://css-tricks.com/data-uris/
```
Comment (pissang) [inline src/component/toolbox/feature/SaveAsImage.ts:79]: It's better to avoid URL referencing like this. It may have license issues

**Comment by pissang** [comment]
Also I'm not sure if we should remove the edge condition here https://github.com/apache/incubator-echarts/blob/a3a0e4582b59dd974d349eef8dfde2b6ebe229ab/src/component/toolbox/feature/SaveAsImage.ts#L61 since edge is using chromium now

File: src/component/toolbox/feature/SaveAsImage.ts (hunk 1 above)
Comment (plainheart) [reply src/component/toolbox/feature/SaveAsImage.ts:79]: Thanks for your reminder. Will remove it.

**Comment by plainheart** [comment]
> I'm not sure if we should remove the edge condition here. edge is using chromium now

@pissang I think we may need to tweak the logic of checking if the browser is Edge in zrender.
Before using chromium core, the latest version of Edge is `18.x`. Now it's increasing from `78.x`.

**Changes in zrender would be:**

```diff
// https://github.com/ecomfe/zrender/blob/master/src/core/env.ts#L58
- const edge = ua.match(/Edge\/([\d.]+)/);
+ const edge = ua.match(/Edge?\/([\d.]+)/); // make `e` optional to match new edge with chromium core.
```

Old UserAgent

```
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.102 Safari/537.36 Edge/18.18363
```

New UserAgent

```
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36 Edg/87.0.664.60
```



**So, the judgment here should be:**

```js
// Chrome, Firefox, New Edge
if (typeof MouseEvent === 'function' && (env.browser.newEdge || (!env.browser.ie && !env.browser.edge))) {
```

The related commit is https://github.com/apache/incubator-echarts/pull/13732/commits/177f1fc5dd489e90eb1c4cf055e342fbc4f6ef32

**Comment by pissang** [comment]
I think it looks good to me.  Pin @100pah to have a further check.

**Review by 100pah — APPROVED** [review]

**Comment by 100pah** [comment]
It looks good to me. @plainheart

**Comment by partitect** [comment]
issue #13655 not resolved

when select the renderer svg, svg export image download but not opening

`option ={
tooltip: {
trigger: "axis"
},
legend: {
data: ["Tictoc", "Whatsapp", "Snapchat"],
itemGap: 30,
itemWidth: 40,
itemHeight: 40
},
toolbox: {
top: 10,
right: 10,
show: true,
feature: {
mark: { show: false },
dataView: { show: false, readOnly: false },
magicType: { show: true, type: ["stack", "tiled"], title: "Change" },
restore: { show: true, title: "Refresh" },
saveAsImage: { show: true, title: "Save As Image",pixelRatio:2,type:'svg' }
}
},
grid: {
show: false,
containLabel: true,
left: "10",
right: "30",
top: "70",
bottom: "10"
},
color: ["#5ea9dd", "#54cc61", "#fffa04"],

calculable: true,
xAxis: [
	{
		type: "category",
		boundaryGap: false,
		data: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
	}
],
yAxis: [
	{
		type: "value"
	}
],
series: [
	{
		name: "Tictoc",
		type: "line",
		smooth: true,
		itemStyle: {
			normal: {
				areaStyle: { type: "default", shadowColor: "rgba(0, 0, 0, 0.9)", shadowBlur: 30, opacity: 0.8 },
				lineStyle: { opacity: 0 }
			}
		},
		data: [1320, 1132, 601, 234, 120, 90, 1680],
		symbol: "image://https://marka-logo.com/wp-content/uploads/2020/04/TikTok-Logo.png",
		symbolSize: 40
	},
	{
		name: "Whatsapp",
		type: "line",
		smooth: true,
		itemStyle: {
			normal: {
				areaStyle: { type: "default", shadowColor: "rgba(0, 0, 0, 0.9)", shadowBlur: 30, opacity: 0.8 },
				lineStyle: { opacity: 0 }
			}
		},
		data: [520, 182, 434, 1420, 390, 30, 1480],
		symbol: "image://https://marka-logo.com/wp-content/uploads/2020/04/TikTok-Logo.png",
		symbolSize: 40
	},
	{
		name: "Snapchat",
		type: "line",
		smooth: true,
		itemStyle: {
			normal: {
				areaStyle: { type: "default", shadowColor: "rgba(0, 0, 0, 0.9)", shadowBlur: 30, opacity: 0.8 },
				lineStyle: { opacity: 0 }
			}
		},

		data: [10, 12, 21, 54, 260, 830, 710],
		symbol: "image://https://marka-logo.com/wp-content/uploads/2020/04/TikTok-Logo.png",
		symbolSize: 40
	}
]
}`

**Pull request merged**